mender.go

```diff
@@ -585,22 +640,46 @@ func (m *mender) TransitionState(to State, ctx *StateContext) (State, bool) {
 		to.SetTransition(from.Transition())
 	}
 
+	report := func() *client.StatusReportWrapper {
+		if shouldReportUpdateStatus(to.Id()) {
+			upd, err := getUpdateFromState(to)
+			if err != nil {
+				log.Error(err)
+				return nil
+			}
+			return &client.StatusReportWrapper{
+				API: m.api.Request(m.authToken),
+				URL: m.config.ServerURL,
+				Report: client.StatusReport{
+					DeploymentID: upd.ID,
+					Status:       to.Id().Status(),
+				},
+			}
+		}
+		return nil
+	}()
```

Comment:
This construct is a bit confusing. If I read it correctly you are both creating and calling the function in the same statement. Can you simply avoid the function altogether and assign directly to `result` without executing `return`?

Reply:
Maybe I was trying to bee too smart here. Fixed!